### PR TITLE
Issue #28 Add travis-ci system

### PR DESCRIPTION
**Pull request by knesenko**
Signed-off-by: Kiril Nesenko <knesenko@vmware.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+go:
+  - 1.7.x
+  - 1.8.x
+
+go_import_path: github.com/vmware/virtual-security-module
+
+install:
+  - make install-deps
+  - go get github.com/mattn/goveralls
```
Comment (asafka) [inline .travis.yml]: doesn't this need to be added to install-deps task? or this is just for travis?

File: Makefile [hunk 2]
```diff
@@ -14,9 +14,17 @@ DOC_DIR := $(PROJECT_DIR)/doc
 DOC := $(DOC_DIR)/swagger.json
 
 GO_ENV := GOPATH=$(GOPATH)
+GO_LINT := PATH=${PATH}:$(GOPATH)/bin golint
```
Comment (asafka) [inline Makefile]: I'm not sure we want to introduce GO_LINT at this point - I had problems with it. Is our current codebase passing lint?

File: Makefile [hunk 3]
```diff
@@ -34,8 +43,21 @@ build: fmt vet
...
+	if [ $$($(GO_LINT) ./... | wc -l) != 0 ]; then \
+	        if $(is_ci); then \
+	                echo "FATAL: you should run 'make lint' on your local system"; \
+	                exit 1; \
+	        fi; \
+	fi
+
 fmt:
-	$(GO) fmt ./...
+	if [ $$($(GO) fmt ./... | wc -l) != 0 ]; then \
```
Comment (asafka) [inline Makefile]: Why this change? don't we want fmt to just silently format the files for us (like it does today)?

File: model/validate.go [hunk 4]
```diff
@@ -7,9 +7,31 @@ import (
 	"net/http"
 	"strings"
 
+	"fmt"
 	"github.com/vmware/virtual-security-module/util"
+	"reflect"
 )
 
+func (s *SecretEntry) Equal(t *SecretEntry) (bool, error) {
```
Comment (asafka) [inline model/validate.go]: returning both bool and error is a bit redundant:
(1) if you want to return an informative error, then you don't need to return a bool, because err == nil if and only if bool == true.
(2) If you just want to know if the two structures are Equal just return a bool.

Personally I would go with the second option here.

File: .travis.yml (hunk 1 above)
Comment (knesenko) [inline .travis.yml]: this is for travis only

File: Makefile (hunk 2 above)
Comment (knesenko) [inline Makefile]: it just a place holder ... didn't enabled it

File: Makefile (hunk 3 above)
Comment (knesenko) [inline Makefile]: we want this to fail on travis if files are not formated

File: model/validate.go (hunk 4 above)
Comment (knesenko) [inline model/validate.go]: will fix
